forge/src/main/java/org/openjdk/skara/forge/github/GitHubPullRequest.java

```diff
@@ -821,4 +827,12 @@ private ZonedDateTime lastMarkedAsReadyTime(JSONValue timelineJSON) {
                 .max(ZonedDateTime::compareTo)
                 .orElseGet(this::createdAt);
     }
+
+    @Override
+    public boolean diffLimited() {
+        if (diffLimited.isEmpty()) {
+            throw new RuntimeException("Diff() has not been called. Can't evaluate if diff is limited");
+        }
+        return diffLimited.get();
+    }
```

Comment:
I think it would be better to add a boolean field `complete` to `Diff`, instead of having this be a property of a PullRequest. Then we avoid this awkward requirement for ordering calls on PullRequest.

Reply:
Yes, this sounds much better. Thank you for the suggestion.